drivers/pinctrl/pinctrl_sy1xx.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 sensry.io
+ */
+
+#define DT_DRV_COMPAT sensry_sy1xx_pinctrl
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+
+#include <zephyr/arch/common/sys_io.h>
+
+#define SY1XX_INVALID_BASE_ADDR (0UL)
+#define SY1XX_INVALID_BASE_SIZE (0UL)
+
+#define SY1XX_PAD_BASE_ADDR(nodelabel)                                                             \
+	COND_CODE_1(DT_NODE_EXISTS(DT_NODELABEL(nodelabel)),					   \
+		(DT_REG_ADDR(DT_NODELABEL(nodelabel))),						   \
+		(SY1XX_INVALID_BASE_ADDR))
```

Comment:
if such node doesn't exist, driver should not even be compiled. Which soc is using this driver, btw?

Reply:
this is the soc's DT https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/riscv/sensry/ganymed-sy1xx.dtsi

should I include the pinctrl in the DTS together with this pull request or create a new pull request for the DTS changes?

Reply:
this is how I would add my pinctrl

https://github.com/zephyrproject-rtos/zephyr/commit/096ceb322af9d5b11de13b732e27f5e4af62c0e5

Reply:
and this is how I want to use it

https://github.com/zephyrproject-rtos/zephyr/commit/3beed2c58e8cc79cf4f753cd68dde6d773674070

Reply:
alright, I made the changes.